core/commands/ls.go

```diff
@@ -78,12 +78,13 @@ it contains, with the following format:
 				Links: make([]LsLink, len(dagnode.Links)),
 			}
 			for j, link := range dagnode.Links {
-				link.Node, err = link.GetNode(req.Context(), node.DAG)
+				var linkNode *merkledag.Node
+				linkNode, err = link.GetNode(req.Context(), node.DAG)
```

Comment:
I fail to see the reason for reusing the err var here:
if an err happens inside the loop, the function returns right away, and if the loop is done, it means the err is `nil` -- same behavior for both reused / scoped errs.